### DISCUSSION TITLE
How to get URL/Origin with which metamask is interacting with for Transaction insights

### DISCUSSION THREAD

**Discussion by yathishram** [MetaMask/snaps#1190] [Q&A] [answered]
Needed to know is it possible to get URL/origin like we get for `onRpcRequest` for `onTransaction` as well. If there's any hack for this do let me know. 

Thanks in advance for the help

**Comment by ziad-saab** [comment, marked as answer]
In a future release of Flask, you'll be able to do this using the `allowTransactionOrigin` caveat. We have a draft of documentation you can look at here:

https://github.com/MetaMask/metamask-docs/pull/605/files#diff-9c416d04d3c9fb8d0e6d16d5ed453668e9508124576e6e73f5dd1e0a43e0a481R89

**Reply by yathishram** [reply]
Oh! this is great!. Is the feature live or when can we expect this to be live?

**Reply by ziad-saab** [reply]
If you build MetaMask Flask from source using the `develop` branch then you'll be able to use that feature. Follow the instructions here:

https://github.com/MetaMask/metamask-extension#building-locally

Make sure to be on the `develop` branch and use `--build-type=flask` to get the developer preview

**Reply by yathishram** [reply]
Awesome! Thanks for the help!
